src/GeniusYield/Test/Utils.hs

```diff
@@ -387,17 +236,20 @@ pattern (:=) x y = (x, y)
 
 infix 0 :=
 
--------------------------------------------------------------------------------
--- Preset StdGen
--------------------------------------------------------------------------------
-
-pureGen :: StdGen
-pureGen = mkStdGen 42
-
-{- -----------------------------------------------------------------------------
-  CLB
------------------------------------------------------------------------------ -}
+{- | Utilizing 'ExceptT' as a "eager monad" transformer.
 
--- | Variant of `logInfo` from @Clb@ that logs a string with @Info@ severity.
-logInfoS :: Monad m => String -> Clb.ClbT m ()
-logInfoS s = Clb.logInfo $ Clb.LogEntry Clb.Info s
+'Left' does not indicate failure, rather it indicates that "target value has been obtained"
+and that we can exit eagerly.
+-}
+type EagerT m a = ExceptT a m ()
+
+-- | If we have a 'Just' value, we can exit with it immediately. So it gets converted
+-- to 'Left'.
+maybeToEager :: Monad m => Maybe a -> EagerT m a
+maybeToEager (Just a) = throwError a
+maybeToEager Nothing  = pure ()
+
+-- If all goes well, we should finish with a 'Left'. if not, we perform the
+-- given action to signal error.
+runEagerT :: Monad m => m a -> ExceptT a m () -> m a
+runEagerT whenError = runExceptT >=> either pure (const whenError)
```

Comment:
Little utility that probably should be in its own module but I only use it in like one or two functions here.

Reply:
https://github.com/geniusyield/atlas/pull/324/files#r1692973125